### DISCUSSION TITLE
How can i determin if a fileEntry returned by filer api is a directory?

### DISCUSSION THREAD

**Discussion by shamork** [seaweedfs/seaweedfs#4650] [Q&A] [answered]
How can i determin if a fileEntry returned by filer api is a directory?

File Entry info shows bellow
{
      "FullPath": "/x/x/x/x/x",
      "Mtime": "2023-07-09T23:20:42+08:00",
      "Crtime": "2023-07-09T23:20:42+08:00",
      "Mode": 2147484153,
      "Uid": 4294967295,
      "Gid": 4294967295,
      "Mime": "",
      "TtlSec": 0,
      "UserName": "",
      "GroupNames": null,
      "SymlinkTarget": "",
      "Md5": null,
      "FileSize": 0,
      "Rdev": 0,
      "Inode": 0,
      "Extended": null,
      "HardLinkId": null,
      "HardLinkCounter": 0,
      "Content": null,
      "Remote": null,
      "Quota": 0
    },

**Comment by kmlebedev** [comment]
If you want a directory then you need to enter the appropriate mime type `httpd/unix-directory`

**Reply by chrislusf** [reply, marked as answer]
The Mode has the directory bit.

**Reply by shamork** [reply]
ok! that is `bool isDir=fileEntry.Mode&0x80000000 !=0;`
